### PR TITLE
perf: added caching to XBlockSkillsViewSet list endpoint

### PR DESCRIPTION
**Pull request by hamza-56**
### [PROD 4135](https://2u-internal.atlassian.net/browse/PROD-4135)

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/openedx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

**Review by Ali-D-Akbar — APPROVED** [review]
Looks good to me.

File: taxonomy/api/v1/views.py [hunk 1]
```diff
@@ -40,6 +41,8 @@
     XBlockSkills,
 )
 
+CACHE_TIMEOUT_XBLOCK_SKILLS_SECONDS = 60 * 60
```
[the remark this answers is not in the document]
Comment (AfaqShuaib09) [reply taxonomy/api/v1/views.py:44]: +1 here

**Review by AfaqShuaib09 — APPROVED** [review]

File: taxonomy/api/v1/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hamza-56) [reply taxonomy/api/v1/views.py]: Updated

**Review by saleem-latif — APPROVED** [review]

**Review by jajjibhai008 — APPROVED** [review]
Could you please add the necessary details along with the ticket link in the PR description? Also, try to resolve the coverage issues before merging.
Other than that, everything looks good to me 👍.

File: taxonomy/api/v1/views.py [hunk 2]
```diff
@@ -320,6 +322,19 @@ def get_queryset(self):
             ),
         ).only('id', 'skills', 'usage_key', 'requires_verification', 'auto_processed', 'hash_content')
 
+    def list(self, request, *args, **kwargs):
+        cache_key = get_cache_key(domain='taxonomy', subdomain='xblock_skills', params=request.query_params)
+
+        cached_response = TieredCache.get_cached_response(cache_key)
+        if cached_response.is_found:
+            return Response(cached_response.value)
```
Comment (DawoudSheraz) [inline taxonomy/api/v1/views.py:330]: We should add a unit test for this.

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**